### PR TITLE
Mobile class pages part 2

### PR DESCRIPTION
**Pull request by puneetjohal**
# Purpose

###### A brief description of the purpose of the PR's changes for the project.
Adds expendables displaying class information and section information.

<br>

# Tickets

###### A link to any tickets this PR is associated with on Trello.

- https://trello.com/c/xYdsNJbM/296-mobile-class-pages-part-2

# Contributors

###### Anyone who contributed to this PR for future reference.

- @puneetjohal 

# Feature List

###### A list of more in-depth and technical changes, additions, deletions, and fixes this PR provides.

- Expendables containing additional information about the class
image

- Section information (expanded)
image

- Section information (collapsed)
image

# Notes

###### A list of miscellaneous notes for this pull request, such as whether we need to add something later for production, or that it was time boxed, etc.

<br>

# Checklist

- [x] Filled out PR template :wink:
- [ ] Approved by designers
- [x] Is passing linting checks
- [x] Is passing tsc
- [x] Is passing existing tests
- [x] Has documentation and comments in code
- [x] Has test coverage for code
- [x] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

File: components/ClassPage/mobile/MobileClassInfoExpandable.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  RecentProfessors: 'RECENT PROFESSORS',
+  RecentSemestersOffered: 'RECENT SEMESTERS OFFERED',
+  NUPaths: 'NUPATHS',
+  Prerequisites: 'PREREQUISITES',
+  Corequisites: 'COREQUISITES',
+  PrerequisiteFor: 'PREREQUISITE FOR',
+  OptionalPreresequisiteFor: 'OPTIONAL PREREQUISITE FOR',
+};
+
+export type MobileClassInfoExpandableProps = {
```
Comment (andrewydai) [inline components/ClassPage/mobile/MobileClassInfoExpandable.tsx]: don't need to export

File: components/ClassPage/mobile/MobileClassInfoExpandable.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        <p>None</p>
+      ) : (
+        <ul>
+          {classPageInfo.class.latestOccurrence.nupath.map((nupath) => {
+            return <li>{nupath}</li>;
+          })}
+        </ul>
+      );
+
+    // TODO: needs a design
```
Comment (andrewydai) [inline components/ClassPage/mobile/MobileClassInfoExpandable.tsx]: Why don't we have branches for the rest of the supported info types? I see some designs on Figma, unsure if I missed something

File: components/ClassPage/mobile/MobilePageContent.tsx [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+                    href={classPageInfo.class.latestOccurrence.prettyUrl}
+                    className="bannerPageLink"
+                  >
+                    <IconGlobe width="22" height="22" className="globe-icon" />
+                    <span>View the course on Banner</span>
+                  </a>
+                </div>
+
+                <div className="notifyButton">
+                  {/* <SignUpForNotifications
```
Comment (andrewydai) [inline components/ClassPage/mobile/MobilePageContent.tsx:233]: Do notifs not work rn?

File: components/ClassPage/mobile/MobilePageContent.tsx [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+                          Math.round(mean(seatsAvailable(classPageInfo)))
+                        ).toFixed(1)}
+                        %
+                      </b>{' '}
+                      of seats per section filled.
+                    </li>
+                  </ul>
+                </div>
+
+                <div className="expandables">
```
Comment (andrewydai) [inline components/ClassPage/mobile/MobilePageContent.tsx:207]: This component is getting pretty monstrous, let's maybe use a map here on the object values, either export an array and map over it or use `Object.values(SupportedInfoTypes).map(() => ...)`

File: components/ClassPage/mobile/MobileClassInfoExpandable.tsx (hunk 2 above)
Comment (andrewydai) [reply components/ClassPage/mobile/MobileClassInfoExpandable.tsx]: Nevermind I understand that this is the next part

**Review by andrewydai — CHANGES_REQUESTED** [review]
Couple small things

**Comment by puneetjohal** [comment]
Added in prereq information. I tried to follow the design as closely as possible, but ultimately chose to simplify it just a little in the interest of time since I was having some trouble parsing out "sequences"/`OR`-ed `AND`s. Here are some examples:

No prereqs (CS2500):
image

One prereq (CS2510):
image

Two `AND`-ed prereqs (CS2810):
image

Nested prereqs (CS3000):
image

The most complex prereq tree I could find among fall classes (CIVE2221):
image

I used the same component for coreqs, so they look the same (just without the nesting).

File: components/ClassPage/mobile/MobilePageContent.tsx (hunk 3 above)
Comment (puneetjohal) [reply components/ClassPage/mobile/MobilePageContent.tsx:233]: Can this be in a separate PR? I'm out of the loop on mobile notif designs, and I think I need to merge in some notif work from master.

**Review by andrewydai — APPROVED** [review]
lgtm!

**Pull request merged**